omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/api/SearchQuery.java

```diff
@@ -94,14 +155,30 @@ public int hashCode() {
 		return result;
 	}
 	
+	/**
+	 * @see Object#equals(Object)
+	 * @param obj the object to test for if equal to this
+	 * @return true if obj is equal to this otherwise false
+	 * @should return true if given this
+	 * @should return true if this optional parameters and required parameters are equal to given
+	 *         search query
+	 * @should be symmetric
+	 * @should be transitive
+	 * @should return false if given null
+	 * @should return false if given an object which is not an instanceof this class
+	 * @should return false if this optional parameters is not equal to the given search queries
+	 *         optional parameters
+	 * @should return false if this required parameters is not equal to the given search queries
+	 *         required parameters
+	 */
 	@Override
 	public boolean equals(Object obj) {
-		if (this == obj)
+		if (this == obj) {
 			return true;
-		if (obj == null)
```

Comment:
yes :) since Im using !(obj instanceof SearchQuery)
null is already covered. null is never an instanceof SearchQuery.